archive_converter.py

```diff
@@ -13,8 +13,57 @@
 # limitations under the License.
 
 import email
+import os
 from setup_gmail import Message
 
+class ArchiveMessageIndex(object):
+    def __init__(self):
+        self._lookup_index = {} # Maps MessageKey to Thread
+        self._messages_seen = {} # Maps message.id to message
+
+    def find_thread(self, key: str) -> Message:
+        return self._lookup_index[key]
+
+    def update(self):
+        fully_updated = False
+        new_messages = []
+
+        for filename in os.listdir('test_data'):
+            if filename.endswith(".txt"):
+                email = generate_email_from_file('test_data/' + filename)
+                if(email.id not in self._messages_seen):
+                    new_messages.append(email)
+        self._add_messages_to_index(new_messages)
+
+    def _add_messages_to_index(self, new_messages):
+        need_parent = []
+        for message in new_messages:
+            self._messages_seen[message.id] = message
+
+            if message.in_reply_to is None:
+                key = message.get_key()
+                print(key)
+                if key:
+                    self._lookup_index[key] = message
+                else:
+                    # If message doesn't have the LKML style [PATCH v4 00/11],
+                    # we don't need to put it in the index.
+                    continue
+            else:
+                need_parent.append(message)
+        for message in need_parent:
+            if message.in_reply_to not in self._messages_seen:
+                # If we don't have the parent message, then we can't reconstruct
```

Comment:
Hmm, seems like we should maybe print something here to signal this rather than silently drop it.

This code isn't using https://docs.python.org/3/library/logging.html, so perhaps just a print statement would be fine.